client/src/partials/enterprises/enterprises.js

```diff
@@ -31,65 +30,161 @@ function EnterpriseController(Enterprises, Currencies, util, Notify) {
     // load enterprises
     Enterprises.read(null, { detailed : 1 })
     .then(function (enterprises) {
-      vm.enterprises = enterprises;
-    }).catch(Notify.handleError);
-
+      vm.hasEnterprise = enterprises.length ? true : false;
+      vm.enterprises = vm.hasEnterprise ? enterprises : [];
+
+      /**
+       * @note: set the enterprise to the first one
+       * this choice need the team point of view for to setting the default enterprise
+       */
+      vm.enterprise = vm.hasEnterprise ? vm.enterprises[0] : {};
+    })
+    .then(Accounts.accounts)
+    .then(function (list) {
+      vm.accounts = list;
+
+      // gain account
+      vm.enterprise.gain_account = vm.accounts.get(vm.enterprise.gain_account_id);
+      // loss account
+      vm.enterprise.loss_account = vm.accounts.get(vm.enterprise.loss_account_id);
+    })
+    .then(refreshProjects)
+    .catch(Notify.handleError);
+
+    // load accounts
+    Accounts.accounts()
+    .then(function (list) {
+      vm.accounts = list;
+
+      // gain account
+      vm.enterprise.gain_account = vm.accounts.get(vm.enterprise.gain_account_id);
+      // loss account
+      vm.enterprise.loss_account = vm.accounts.get(vm.enterprise.loss_account_id);
+    })
+    .catch(Notify.handleError);
+
+    // load currencies
     Currencies.read()
     .then(function (currencies) {
       vm.currencies = currencies;
     }).catch(Notify.handleError);
 
-    vm.view = 'default';
-  }
-
-  function cancel() {
-    vm.view = 'default';
-  }
-
-  function create() {
-    vm.view = 'create';
-    vm.enterprise = {};
-  }
-
-  // switch to update mode
-  function update(id) {
-    vm.enterprise = id;
-    vm.view = 'update';
-  }
-
-  // refresh the displayed Enterprises
-  function refreshEnterprises() {
-    return Enterprises.read(null, { detailed : 1 })
-    .then(function (enterprises) {
-      vm.enterprises = enterprises;
-    });
   }
 
   // form submission
   function submit(form) {
+
     if (form.$invalid) {
       Notify.danger('FORM.ERRORS.HAS_ERRORS');
       return;
     }
 
     var promise;
-    var creation = (vm.view === 'create');
-    var enterprise = angular.copy(vm.enterprise);
+    var creation = (vm.hasEnterprise === false);
+    var enterprise = cleanEnterpriseForm(vm.enterprise);
 
     promise = (creation) ?
       Enterprises.create(enterprise) :
       Enterprises.update(enterprise.id, enterprise);
 
     return promise
       .then(function () {
-        return refreshEnterprises();
-      })
-      .then(function () {
-        update(enterprise.id);
-        Notify.success(creation ? 'FORM.INFO.SAVE_SUCCESS' : 'FORM.INFO.UPDATE_SUCCESS');
+        Notify.success(creation ? 'FORM.INFOS.SAVE_SUCCESS' : 'FORM.INFOS.UPDATE_SUCCESS');
       })
       .catch(Notify.handleError);
   }
 
+  function cleanEnterpriseForm(enterprise) {
```

Comment:
@sfount implemented a utility for doing this, `util.filterFormElements()`.  You should ask him how it works and see if it is appropriate here.